### ISSUE TITLE
TextTracks not populated and they can't be mocked

### ISSUE THREAD

**Issue by DaveStein** [jsdom/jsdom#2541]
### Basic info:
My production code has the line `videoTag.textTracks[0].mode = isOn ? 'showing' : 'hidden'`. When I am using `jsdom`, `textTracks` does not get populated. If I try to set `textTracks` manually as a mock, I get `TypeError: Cannot set property textTracks of #<HTMLMediaElement> which has only a getter`

- **Node.js version:** 8.15.0
- **jsdom version:** 14.0.0

### Minimal reproduction case

```js
const { JSDOM } = require("jsdom");

const dom = new JSDOM(`<!DOCTYPE html>`);
const doc = dom.window.document;
const videoTag = doc.createElement('video');
doc.body.appendChild(videoNode);


const trackTag = doc.createElement('track');
trackTag.setAttribute('default', '');
trackTag.setAttribute('src', `/captions`);
videoTag.appendChild(trackTag);

console.log(videoTag.textTracks); // empty array

```

### How does similar code behave in browsers?

textTracks is populated when the track tag is appended.

**Comment by DaveStein** [comment]
I also did a sanity check, that even if the `src` goes to nowhere, the `videoTag` still gets a `TextTrack`

**Comment by domenic** [comment]
This is indeed not an implemented feature yet.

In the meantime, you can mock using `Object.defineProperty` to override the property, instead of trying to assign to a getter-only property.

**Comment by DaveStein** [comment]
This was an impressive response time :)

**Comment by DaveStein** [comment]
Got weird behavior here?

```
videoNode = doc.createElement('video');
console.log(videoNode.textTracks); // []
Object.defineProperty(videoNode, 'textTracks', [{
  mode: 'disabled'
}]);
console.log(videoNode.textTracks); // undefined
```

**Comment by DaveStein** [comment]
@domenic am I being the dumb? :)

**Comment by domenic** [comment]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty

**Comment by DaveStein** [comment]
@domenic Ah yes, I was being dumb. My memory had me thinking I knew how to use this, but apparently I needed to do:

```
    const textTracks = [
      {
        mode: 'disabled'
      }
    ];
    Object.defineProperty(videoNode, 'textTracks', { value: textTracks });
```

Should we leave this open as a feature request then? And this comment is the workaround?

**Comment by domenic** [comment]
Yep, that sounds about right. One day maybe we'll implement text tracks, but until then, you can mock it like all other things in jsdom.

**Comment by jony89** [comment]
+1 for implementing `TextTrackCueList` & `TextTracks`

**Comment by LeBenLeBen** [comment]
Is this the reason why `window.TextTrack` is undefined or should I create another issue for that?

**Comment by MacgyverMartins** [comment]
I am testing a component with video and get the same error when write a Jest test:
```bash
Error: Uncaught [ReferenceError: TextTrack is not defined]
```

**Comment by Viktor19931** [comment]
in test you could just mock lib jest.mock('plyr');
